rabbitmq/persistent_consumer.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2019 SumUp Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rabbitmq
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/sumup-oss/go-pkgs/logger"
+)
+
+type PersistentConsumer struct {
+	consumer          *Consumer
+	mu                sync.Mutex
+	reconnectTimeout  time.Duration
+	rabbitClientCfg   *ClientConfig
+	rabbitClientSetup *Setup
+}
+
+func NewPersistentConsumer(
+	client *RabbitMQClient,
+	setup *Setup,
+	handler Handler,
+	logger logger.StructuredLogger,
+	metric Metric,
+	cfg ConsumerConfig,
+	reconnectTimeout time.Duration,
+) *PersistentConsumer {
+	consumer := &Consumer{
+		client:  client,
+		handler: handler,
+		logger:  logger,
+		metric:  metric,
+		cfg:     cfg,
+		stopWg:  sync.WaitGroup{},
+	}
```

Comment:
Any reason not to use the `NewConsumer` function here? The only difference I guess is that it sets the wait group rather than us setting it here explicitly https://github.com/sumup-oss/go-pkgs/blob/master/rabbitmq/consumer.go#L48-L63?

Reply:
Yeah, probably will be prettier that way. Added the NewConsumer func.